### ISSUE TITLE
explicitly state full integer types.

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#194]
@bartnijssen has suggested (in #188) that we be explicitly state the full datatype when typecasting or declaring integer variables.  For example:
```C
    unsigned short             day;
    double                     sec;
    size_t                     rec;
    int                        idx;
    unsigned                   uidx;
```
would become:
```C
    unsigned short int         day;
    double                     sec;
    size_t                     rec;
    int                        idx;
    unsigned int               uidx;
```

This table outlines the synonymous datatypes in C: http://en.wikipedia.org/wiki/C_data_types#Basic_types

@bartnijssen, would you suggest that we use the most explicit form?  (`int` -> `singed int`, `unsigned` -> `unsigned int`, etc.).  Or, where you mostly hoping to add int to the tail end of `unsigned` and `short`?

**Comment by bartnijssen** [comment]
The latter - just to remain consistent with the rest of VIC. I have never seen anyone actually use `signed`, but not including `int` after `unsigned` is potentially confusing within the context of VIC. 

Note that the goal is not simply to adhere to the most recent version of the C standard. Code in general is much easier to read and maintain if it is self-consistent. For SUMMA we actually came up with a style guide. We do not have that for VIC, but I feel strongly that code maintenance is facilitated by being consistent within the existing code.

**Comment by bartnijssen** [comment]
Closed per #200
